src/catalog/src/system_schema/pg_catalog/pg_catalog_memory_table.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use datatypes::schema::{ColumnSchema, Schema, SchemaRef};
+use datatypes::vectors::{Int16Vector, StringVector, UInt32Vector, VectorRef};
+
+use super::oid_column;
+use super::table_names::PG_TYPE;
+use crate::memory_table_cols;
+use crate::system_schema::memory_table::tables::{i16_column, string_column};
+fn pg_type_schema_columns() -> (Vec<ColumnSchema>, Vec<VectorRef>) {
+    memory_table_cols!(
+        [oid, typname, typlen],
+        [
+            (1, "String", -1),
+            (2, "Binary", -1),
+            (3, "Int8", 1),
+            (4, "Int16", 2),
+            (5, "Int32", 4),
+            (6, "Int64", 8),
+            (7, "UInt8", 1),
+            (8, "UInt16", 2),
+            (9, "UInt32", 4),
+            (10, "UInt64", 8),
+            (11, "Float32", 4),
+            (12, "Float64", 8),
+            (13, "Decimal", 16),
+            (14, "Date", 4),
+            (15, "DateTime", 8),
+            (16, "Timestamp", 8),
+            (17, "Time", 8),
+            (18, "Duration", 8),
+            (19, "Interval", 16),
+            (20, "List", -1),
```

Comment:
I'm not quite familiar with our type system. How do we keep this list updated? Is there any test cases for preventing regressions?

Reply:
Maybe we could acquire this data from the trait [DataType](https://github.com/GreptimeTeam/greptimedb/blob/7ad248d6f6645d898bcd38d19fd8b4bc29462500/src/datatypes/src/types/primitive_type.rs#L352)